core/src/main/java/org/frankframework/receivers/Receiver.java

```diff
@@ -1325,7 +1327,7 @@ private Message processMessageInAdapter(MessageWrapper<M> messageWrapperOriginal
 					statusMessage = t.getMessage();
 					if (pipeLineResult==null) {
 						pipeLineResult=new PipeLineResult();
-						pipeLineResult.setExitCode(500); // If there was an exception that was not handled by the pipeline, consider it an internal server error.
+						pipeLineResult.setExitCode(session.get(PipeLineSession.EXIT_CODE_CONTEXT_KEY, 500)); // If there was an exception that was not handled by the pipeline, consider it an internal server error.
```

Comment:
Default exit-code is still 500 but now we also allow the pipeline to set the exit-code, for instance via the exception-pipe.